internal/game/riser/riser.go

```diff
@@ -319,16 +358,40 @@ func (r *Riser) Update() {
 		} else {
 			r.State = MovingLeft
 		}
-	} else if canPull && actionPressed {
-		if r.World.Player.Rect.Center().X < r.Entity.Rect.Center().X {
-			r.State = MovingLeft
-		} else {
-			r.State = MovingRight
+	} else if canPull && actionPressed { // Surely this code can be better improved
```

Comment:
Maybe just add a:

centerDelta := r.World.Player.Rect.Center().X - r.Entity.Rect.Center().X

and then use that everywhere.